packages/widget-chat/src/components/activity-item-post/index.js

```diff
@@ -2,22 +2,55 @@ import React, {PropTypes} from 'react';
 import classNames from 'classnames';
 
 import Avatar from '../avatar';
-import ActivityItemPostActions from '../activity-item-post-actions';
+import ActivityItemPostAction from '../activity-item-post-action';
 import styles from './styles.css';
 
+import {ICON_TYPE_DELETE, ICON_TYPE_FLAGGED_OUTLINE} from '../icon';
 
 export default function ActivityItemPost(props) {
+  function handleOnDelete() {
+    const {id, onActivityDelete} = props;
+    onActivityDelete(id);
+  }
+
+  function handleOnFlag() {
+    const {id, onActivityFlag} = props;
+    onActivityFlag(id);
+  }
+
+  function wrapActionItem(action, highlight) { // eslint-disable-line react/no-multi-comp
+    const actionClassNames = [`activity-post-action`, styles.activityPostAction];
+    if (highlight) {
+      actionClassNames.push(`activity-post-action-highlighted`, styles.activityPostActionHighlighted);
+    }
+    return (
+      <div className={classNames(actionClassNames)}>
+        {action}
+      </div>
+    );
+  }
+
   const {
     content,
-    id,
     isAdditional,
+    isFlagged,
     isSelf,
     name,
-    onActivityDelete,
     timestamp
   } = props;
 
-  const showDeleteAction = isSelf;
+  let deleteAction;
+  if (isSelf) {
+    deleteAction = <ActivityItemPostAction iconType={ICON_TYPE_DELETE} onClick={handleOnDelete} title="Delete this message" />;
+  }
+  else {
+    deleteAction = <div className={classNames(`action-spacer`, styles.actionSpacer)} />;
+  }
+  deleteAction = wrapActionItem(deleteAction, false);
+
+  let flagAction = <ActivityItemPostAction iconType={ICON_TYPE_FLAGGED_OUTLINE} onClick={handleOnFlag} title="Flag this message" />;
+  flagAction = wrapActionItem(flagAction, isFlagged);
+
```

Comment:
The reason I had to do it this way is that flagged items always display the flag action button, but the delete button only displays on hover. This was the only way I could get it to work with css-modules due to the `:hover` state being on the entire activity item.

I did attempt passing a `isHovered` prop that would do the display instead of css, but it seemed to make it even more complex.